ffi_how_to.md

```diff
@@ -0,0 +1,115 @@
+This document aims to create a detailed work plan for shipping the `pre-built-ffi` workstream.
+The high level approach for this has already been well documented at https://hackmd.io/@mvdan/Hy7iK0TEY.
+
+### Changes to https://github.com/filecoin-project/filecoin-ffi
+
+Going ahead, except for cases where users want to build `filecoin-ffi` from source, `filecoin-ffi` will essentially act as a "wrapper" that forwards all API calls to the corresponding `prebuilt-ffi` module. Since a `prebuilt-ffi` module contains platform-dependent assets (such as static C libraries), APIs in `filecoin-ffi` will need to delegate calls to the "platform-specific" `prebuilt-ffi` module.
+
+Currently, `filecoin-ffi` supports multiple (GOOS + GOARCH) combinations. Therefore, we must modify every public API in `filecoin-ffi` to have a variant for each (GOOS + GOARCH) combination. These variants will delegate the call to the corresponding platform-specific `prebuilt-ffi` module instead of calling into the CGO bindings as they do today (except when users build `filecoin-ffi` from source).
```

Comment:
https://github.com/filecoin-project/filecoin-ffi/blob/03b9503994a1cebe6e1bd333b9f566a7fdb24b4b/.circleci/config.yml#L187-L204 currently only 3 combinations.

And part of this work should be that there remains a well-documented pathway to building and using from source. I imagine that option is going to be more difficult than it is to day since makefiles and bash scripts can determine for you whether there's a binary or not, but this new pathway is just going to result in `go install` failing if you're outside of the common combination range we support.

Reply:
@rvagg 
Build from source should continue working as expected because we will retain all the existing files as it with their corresponding built tags. We will create new files/APIs that depend on the corresponding `prebuilt-ffi-*` modules but those files will explicitly have a !ffi_source built tag to tell the compiler to not use them when building from source.